### DISCUSSION TITLE
Cancel the deprecation of GitHub and other custom git sources

### DISCUSSION THREAD

**Discussion by deivid-rodriguez** [ruby/rubygems#5138] [General]
We were planning to deprecate `:github` and other custom git sources in favor of "do it in your own Gemfile" solutions. However, I think `GitHub` is so common that it's really handy that bundler provides it by default. A while ago I postponed the deprecation to after bundler 3, because I thought it would create too much friction with users for bundler 3. But now I find myself accepting new features for it, and had even forgotten we still plan to deprecate it later.

My proposal is to stick to the custom git sources that we currently have. They're not too hard to maintain and they are very useful for the community. If we get a new feature proposal in the future for some new uncommon source control website, we may want to reject that but I think it's wise to keep what we have now.

**Comment by hsbt** [comment]
I agreed your proposal. There is no strong reason to remove it.

**Comment by deivid-rodriguez** [comment]
I proposed a PR: https://github.com/rubygems/rubygems/pull/5147.

**Comment by duckinator** [comment]
This was resolved by #5147 in December 2021.
